test/controllers/RecordsControllerSpec.scala

```diff
@@ -2,36 +2,80 @@ package controllers
+import io.circe.Printer
+import io.circe.generic.auto._
+import io.circe.syntax._
 import org.scalatest.Matchers._
 import play.api.test.FakeRequest
 import play.api.test.Helpers.{status => playStatus, _}
+import services.ConsignmentService
 import util.FrontEndTestHelper
 
 import scala.concurrent.ExecutionContext
 
 class RecordsControllerSpec extends FrontEndTestHelper {
   implicit val ec: ExecutionContext = ExecutionContext.global
 
+  val totalFiles: Int = 15
+  val consignmentId: UUID = UUID.fromString("b5bbe4d6-01a7-4305-99ef-9fce4a67917a")
+
+  val wiremockServer = new WireMockServer(9006)
+
+  override def beforeEach(): Unit = {
+    wiremockServer.start()
+  }
+
+  override def afterEach(): Unit = {
+    wiremockServer.resetAll()
+    wiremockServer.stop()
+  }
+
   "RecordsController GET" should {
 
-    "render the records page after initial upload is complete and user clicks continue" in {
-      val controller = new RecordsController(getAuthorisedSecurityComponents,
-        new GraphQLConfiguration(app.configuration), getValidKeycloakConfiguration)
-      val recordsPage = controller.recordsPage(UUID.randomUUID()).apply(FakeRequest(GET, "/consignment/123/records"))
+    "render the records page with progress bar after initial upload is complete and user clicks continue" in {
+      val graphQLConfiguration = new GraphQLConfiguration(app.configuration)
+      val consignmentService = new ConsignmentService(graphQLConfiguration)
+      val client = new GraphQLConfiguration(app.configuration).getClient[fileCheck.Data, fileCheck.Variables]()
+
+      val filesProcessed = 1
+      val antivirusProgress = fileCheck.GetConsignment.FileChecks.AntivirusProgress(filesProcessed);
+      val fileChecks = fileCheck.GetConsignment.FileChecks(antivirusProgress)
+      val data: client.GraphqlData = client.GraphqlData(Some(fileCheck.Data(Some(fileCheck.GetConsignment(totalFiles, fileChecks)))))
+      val dataString: String = data.asJson.printWith(Printer(dropNullValues = false, ""))
+
+      wiremockServer.stubFor(post(urlEqualTo("/graphql"))
+        .willReturn(okJson(dataString)))
+
+      val recordsController = new RecordsController(
+        getAuthorisedSecurityComponents,
+        new GraphQLConfiguration(app.configuration),
+        getValidKeycloakConfiguration,
+        consignmentService
+      )
+
+      val recordsPage = recordsController.recordsPage(consignmentId).apply(FakeRequest(GET, s"consignment/${consignmentId}/records"))
+
       playStatus(recordsPage) mustBe OK
       contentType(recordsPage) mustBe Some("text/html")
-      contentAsString(recordsPage) must include ("records.header")
-      contentAsString(recordsPage) must include ("records.title")
+      contentAsString(recordsPage) must include("checkingRecords.header")
+      contentAsString(recordsPage) must include("checkingRecords.title")
+      contentAsString(recordsPage) must include("progress")
```

Comment:
I was thinking about that, I've managed to get it testing that too now. It's currently a bit ugly though so I'll look into how to refactor it to be a bit nicer to read/understand!

Reply:
Yeah, I kind of expected PlaySpec to give us an easier way to inspect the HTML, but I don't think there's a built-in way.

The new assertion is a bit fragile, since the test will fail if someone does something like adding a new class to the element, but I think it's the best option for now.